webapp/src/index.tsx

```diff
@@ -22,8 +25,28 @@ export default class Plugin {
                 const state = store.getState();
                 const post = getPost(state, postID);
 
-                // Don't show up if the post has no attachments. Permissions are checked server-side.
-                return typeof post.file_ids?.length !== 'undefined' && post.file_ids?.length > 0;
+                // Check if post is editable
+                const config = getConfig(state);
+                const edit_time_limit : number = config.PostEditTimeLimit ? Number(config.PostEditTimeLimit) : -1;
+                if (!(edit_time_limit !== -1 && post.create_at + (edit_time_limit * 1000) > Date.now())) {
+                    return false;
+                }
+
+                // Check if post has attachments
+                if (!(typeof post.file_ids?.length !== 'undefined' && post.file_ids?.length > 0)) {
+                    return false;
+                }
+
+                // Check if the user has permissions to edit his own post or edit other's posts if not the author
+                const user = getCurrentUser(state);
+                let permission = 'edit_post';
+                if (post.user_id !== user.id) {
+                    permission = 'delete_others_posts';
+                }
+                return haveIChannelPermission(state, {
+                    channel: post.channel_id,
+                    permission,
+                });
```

Comment:
Works with channel permission but not those with team permission.

Reply:
We are not checking team permissions on the backend either, so this shouldn't be working before either. let me make the checks on par.

Reply:
Thanks all good now!